File: frontend/src/models/BrandGuidelines.ts
// models/BrandGuidelines.ts
import { Schema } from "mongoose";

export interface IBrandGuidelines {
  // Logo Analysis
  logoDescription: string;
  logoColors: string[];
  logoStyle: string;

  // Color System
  colors: {
    primary: string;
    secondary: string;
    accent: string[];
    neutral: string[];
    gradients?: string[];
  };

  // Typography
  fonts: {
    heading: string;
    body: string;
    suggested: string[];
  };

  // Brand Identity
  personality: string[];
  style: string;
  mood: string[];

  // Design Rules
  visualElements: string[];
  designDos: string[];
  designDonts: string[];

  // Design Tokens
  spacing: {
    baseUnit: string;
    scale: number[];
  };
  borderRadius: string[];

  // AI-Generated Content
  suggestedTagline?: string;
  enhancedDescription?: string;
  suggestedKeywords?: string[];
  brandValues?: string[];
  suggestedMission?: string;
  suggestedVision?: string;

  generatedAt: Date;
}

export const BrandGuidelinesSchema = new Schema<IBrandGuidelines>(
  {
    logoDescription: { type: String, required: true },
    logoColors: [{ type: String }],
    logoStyle: { type: String },

    colors: {
      primary: { type: String, required: true },
      secondary: { type: String, required: true },
      accent: [{ type: String }],
      neutral: [{ type: String }],
      gradients: [{ type: String }],
    },

    fonts: {
      heading: { type: String },
      body: { type: String },
      suggested: [{ type: String }],
    },

    personality: [{ type: String }],
    style: { type: String },
    mood: [{ type: String }],

    visualElements: [{ type: String }],
    designDos: [{ type: String }],
    designDonts: [{ type: String }],

    spacing: {
      baseUnit: { type: String },
      scale: [{ type: Number }],
    },
    borderRadius: [{ type: String }],

    suggestedTagline: { type: String },
    enhancedDescription: { type: String },
    suggestedKeywords: [{ type: String }],
    brandValues: [{ type: String }],
    suggestedMission: { type: String },
    suggestedVision: { type: String },

    generatedAt: { type: Date, default: Date.now },
  },
  { _id: false }
);

File: ai-service/app/models/brand_guidelines.py
# app/models/brand_guidelines.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class ColorSystem(BaseModel):
    primary: str
    secondary: str
    accent: List[str]
    neutral: List[str]
    gradients: Optional[List[str]] = []

class Fonts(BaseModel):
    heading: str
    body: str
    suggested: List[str]

class Spacing(BaseModel):
    baseUnit: str
    scale: List[float]

class BrandGuidelines(BaseModel):
    # Logo Analysis
    logoDescription: str
    logoColors: List[str]
    logoStyle: str
    
    # Color System
    colors: ColorSystem
    
    # Typography
    fonts: Fonts
    
    # Brand Identity
    personality: List[str]
    style: str
    mood: List[str]
    
    # Design Rules
    visualElements: List[str]
    designDos: List[str]
    designDonts: List[str]
    
    # Design Tokens
    spacing: Spacing
    borderRadius: List[str]
    
    # AI-Generated Content
    suggestedTagline: Optional[str] = None
    enhancedDescription: Optional[str] = None
    suggestedKeywords: Optional[List[str]] = []
    brandValues: Optional[List[str]] = []
    suggestedMission: Optional[str] = None
    suggestedVision: Optional[str] = None
    
    generatedAt: datetime = datetime.now()
